github.go

```diff
@@ -0,0 +1,267 @@
+package checkup
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+)
+
+var errFileNotFound = fmt.Errorf("file not found on github")
+
+// GitHub is a way to store checkup results in a GitHub repository.
+type GitHub struct {
+	AccessToken     string `json:"access_token"`
+	RepositoryOwner string `json:"repository_owner"`
+	RepositoryName  string `json:"repository_name"`
+	CommitterName   string `json:"committer_name"`
+	CommitterEmail  string `json:"committer_email"`
+	Branch          string `json:"branch"`
+	Dir             string `json:"dir"`
+
+	// Check files older than CheckExpiry will be
+	// deleted on calls to Maintain(). If this is
+	// the zero value, no old check files will be
+	// deleted.
+	CheckExpiry time.Duration `json:"check_expiry,omitempty"`
+
+	client *github.Client
+}
+
+func (gh *GitHub) ensureClient() error {
```

Comment:
For consistency, methods should all have the same receivers: since at least one of them (this one) needs a pointer, probably best to make all of them pointers. So instead of `(gh GitHub)` receivers, we'd have `(gh *GitHub)` across the board.